demo

```diff
@@ -62,16 +80,144 @@ if [[ "$?" -eq 1 ]]; then
     docker-compose pull ${compose_args}
 fi
 
-if [[ "$1" == "up" ]]; then
-    echo "Starting Mender demo environment.."
-fi
-
 # Pass this value on to the GUI container as an env variable
 export INTEGRATION_VERSION=$(git describe --tags --abbrev=0)
 
-exec docker-compose \
-     -f docker-compose.yml \
-     -f docker-compose.storage.minio.yml \
-     -f docker-compose.demo.yml \
-     $CLIENT \
-     "$@"
+if [[ "${!#}" != "up" ]]; then
+    # exec steals the shell, so unless docker-compose is not found,
+    # exit 1 will never happen.
+    exec docker-compose \
+         -f docker-compose.yml \
+         -f docker-compose.storage.minio.yml \
+         -f docker-compose.demo.yml \
+         $CLIENT \
+         "$@"
+    exit 1
+fi
+
+# ------------------------------------------------------------------------------
+#
+#       The following code will only be run in the case ./demo up [args]
+#
+# ------------------------------------------------------------------------------
+
+DOCKER_COMPOSE_PROJECT_NAME="${DOCKER_COMPOSE_PROJECT_NAME:-integration}"
+
+echo "Docker_compose_proj name"
+echo
+echo "name " ${DOCKER_COMPOSE_PROJECT_NAME}
+echo
+
+# Make sure that the demo environment is brought down on SIGINT
+exitfunc() {
+    retval=$(docker-compose \
+         -f docker-compose.yml \
+         -f docker-compose.storage.minio.yml \
+         -f docker-compose.demo.yml \
+         $CLIENT \
+         down)
+    exit $retval
+}
+
+trap exitfunc SIGINT
+
+
+echo "Starting the Mender demo environment..."
+
+docker-compose \
+    -f docker-compose.yml \
+    -f docker-compose.storage.minio.yml \
+    -f docker-compose.demo.yml \
+    $CLIENT \
+    "$@" -d
+
+USER='mender-demo@example.com'
+PASSWORD=$(< /dev/urandom tr -dc A-Za-z0-9 | head -c12)
```

Comment:
Acceptible generation of random passwords?

Reply:
:+1:

No need to connect to a quantum randomness source for this, it's just a demo. ;-)